storm-core/src/clj/backtype/storm/daemon/worker.clj

```diff
@@ -137,14 +138,11 @@
   check highWaterMark and lowWaterMark for backpressure"
   (disruptor/disruptor-backpressure-handler
     (fn []
-      "When worker's queue is above highWaterMark, we set its backpressure flag"
-      (if (not @(:backpressure worker))
-        (do (reset! (:backpressure worker) true)
-            (WorkerBackpressureThread/notifyBackpressureChecker (:backpressure-trigger worker)))))  ;; set backpressure no matter how the executors are
+       (reset! (:transfer-backpressure worker) true)
+       (WorkerBackpressureThread/notifyBackpressureChecker (:backpressure-trigger worker)))
```

Comment:
Will this cause the WorkerBackpressureThread gets too frequently notified even when it is not necessary to do so?
(In most cases, both the :backpressure and :transfer-backpressure have not changed, we may not really need to notify the backpressure-thread to check all executors's flags, which can be additional overhead.)